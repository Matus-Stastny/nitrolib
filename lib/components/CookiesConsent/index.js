"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _CookiesBanner = _interopRequireDefault(require("./components/CookiesBanner"));

var CookiesConsent =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2.default)(CookiesConsent, _React$PureComponent);

  function CookiesConsent() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, CookiesConsent);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(CookiesConsent)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "state", {
      accepted: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleAccept", function () {
      var onAccept = _this.props.onAccept;

      _this.setState({
        accepted: true
      });

      onAccept();
    });
    return _this;
  }

  (0, _createClass2.default)(CookiesConsent, [{
    key: "render",
    value: function render() {
      var accepted = this.state.accepted;

      if (accepted) {
        return null;
      }

      return React.createElement("section", {
        "data-test": "CookiesConsent"
      }, React.createElement(_CookiesBanner.default, {
        onAccept: this.handleAccept
      }));
    }
  }]);
  return CookiesConsent;
}(React.PureComponent);

var _default = CookiesConsent;
exports.default = _default;