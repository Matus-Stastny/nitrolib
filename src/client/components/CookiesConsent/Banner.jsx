// @flow strict
import * as React from "react";
import styled from "styled-components";

import Text from "client/components/Text";

import { brandDefault } from "client/records/Brand";
import mq from "client/services/utils/mediaQuery";

const Container = styled.div`
  display flex;
  justify-content space-between;
  align-items center;
  box-sizing border-box;
  width: 100%;
  padding 10px 20px;
  font-size 12px;
  background ${({ theme }) => theme.colors["grey-900"]};
  color white;
  opacity .94;
  transition bottom .25s ease-in;
  box-shadow 0 -1px 6px 0 rgba(0, 0, 0, .20);

  ${mq.gtTablet`
    padding 20px;
    color ${({ theme }) => theme.colors["grey-800"]};
    background white;
  `}
`;

Container.defaultProps = {
  theme: brandDefault.theme,
};

const Message = styled.p`
  opacity .5;

  ${mq.gtTablet`
    opacity: 1;
  `};
`;

const InfoLink = styled.a`
  text-decoration underline;
  cursor pointer;

  &:hover {
    color white;
    opacity 1;
  }

  ${mq.gtTablet`
    &:hover {
      color initial;
    }
  `};
`;

const AcceptButton = styled.a`
  padding 10px;
  padding-left 20px;
  font-weight bold;
  white-space nowrap;
  cursor pointer;
`;

type Props = {|
  showInfo: () => void,
  accept: () => void,
|};

const Banner = ({ showInfo, accept }: Props) => (
  <Container>
    <Message>
      <Text t={__("content.cookies.popup.full")} />
      &nbsp;
      <InfoLink onClick={showInfo} data-test="CookiesConsent-ShowInfo">
        <Text t={__("common.learn_more")} />
      </InfoLink>.
    </Message>
    <AcceptButton onClick={accept} data-test="CookiesConsent-Accept">
      <Text t={__("common.got_it")} />
    </AcceptButton>
  </Container>
);

export default Banner;
