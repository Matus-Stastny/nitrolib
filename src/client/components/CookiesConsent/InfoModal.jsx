// @flow strict
import * as React from "react";
import styled from "styled-components";

import Modal from "client/components/Modal";
import Text from "client/components/Text";
import { Consumer as BrandConsumer } from "client/services/brand/context";

const Container = styled.div``;

const Title = styled.h1``;

const Content = styled.p``;

const tKeys = {
  kayakwhite: __("content.cookies.kayakwhite"),
  kiwicom: __("content.cookies.kiwicom"),
  paragraphs: __("content.cookies.paragraphs"),
};

type Props = {|
  onClose: () => void,
|};

const InfoModal = ({ onClose }: Props) => (
  <Modal onClose={onClose}>
    <Container>
      <Title>
        <Text t={__("content.pages.cookies.title")} />
      </Title>
      <Content>
        <BrandConsumer>
          {brand => <Text t={tKeys[brand.content.legal.cookies.t_key]} html />}
        </BrandConsumer>
      </Content>
    </Container>
  </Modal>
);

export default InfoModal;
