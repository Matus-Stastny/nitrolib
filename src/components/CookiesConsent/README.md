# CookiesConsent

Closeable cookies thingie.

**Selectors:**
* Container: `[data-test="CookiesConsent"]`
