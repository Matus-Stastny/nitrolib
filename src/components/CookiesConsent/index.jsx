// @flow strict
import * as React from "react";

import CookiesBanner from "./components/CookiesBanner";

type Props = {|
  accepted: boolean, // eslint-disable-line react/no-unused-prop-types
  onAccept: () => void,
|};

type State = {|
  accepted: boolean,
|};

class CookiesConsent extends React.PureComponent<Props, State> {
  state = {
    accepted: false,
  };

  static getDerivedStateFromProps(props: Props) {
    return {
      accepted: props.accepted,
    };
  }

  handleAccept = () => {
    const { onAccept } = this.props;
    this.setState({ accepted: true });

    onAccept();
  };

  render() {
    const { accepted } = this.state;

    if (accepted) {
      return null;
    }

    return <CookiesBanner onAccept={this.handleAccept} />;
  }
}

export default CookiesConsent;
