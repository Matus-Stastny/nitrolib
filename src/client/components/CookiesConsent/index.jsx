// @flow strict
import * as React from "react";

import ClientOnly from "client/components/ClientOnly";
import * as cookiesStore from "client/components/CookiesConsent/services/store";
import Banner from "./Banner";
import InfoModal from "./InfoModal";

type State = {|
  isAccepted: boolean,
  infoShown: boolean,
|};

// TODO: Show consent only in Europe
class CookiesConsent extends React.PureComponent<{}, State> {
  state: State = {
    isAccepted: cookiesStore.isAccepted(),
    infoShown: false,
  };

  handleAccept = () => {
    this.setState({ isAccepted: true });

    cookiesStore.saveAccepted();
  };

  handleShowInfo = () => {
    this.setState({ infoShown: true });
  };

  handleHideInfo = () => {
    this.setState({ infoShown: false });
  };

  render() {
    const { isAccepted, infoShown } = this.state;

    return (
      <ClientOnly>
        {!isAccepted && ( // TODO: Add continent check
          <React.Fragment>
            <Banner onAccept={this.handleAccept} onShowInfo={this.handleShowInfo} />
            {infoShown && <InfoModal onClose={this.handleHideInfo} />}
          </React.Fragment>
        )}
      </ClientOnly>
    );
  }
}

export default CookiesConsent;
