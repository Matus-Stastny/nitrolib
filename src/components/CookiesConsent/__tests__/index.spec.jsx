// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import CookiesConsent from "../index";

describe("#CookiesConsent", () => {
  test("render - cookies accepted", () => {
    const wrapper = shallow(<CookiesConsent accepted onAccept={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("render - cookies not accepted", () => {
    const wrapper = shallow(<CookiesConsent accepted={false} onAccept={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("accept cookies", () => {
    const onAccept = jest.fn();
    const wrapper = shallow(<CookiesConsent accepted onAccept={onAccept} />);

    wrapper.instance().handleAccept();

    expect(onAccept).toBeCalled();
  });
});
