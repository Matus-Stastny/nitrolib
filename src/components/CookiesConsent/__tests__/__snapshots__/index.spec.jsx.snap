// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CookiesConsent render - cookies accepted 1`] = `
<CookiesBanner
  onAccept={[Function]}
/>
`;

exports[`#CookiesConsent render - cookies not accepted 1`] = `
<CookiesBanner
  onAccept={[Function]}
/>
`;
